src/profile/profile.controller.ts

```diff
@@ -163,4 +165,23 @@ export class ProfileController {
       excludeExtraneousValues: true
     });
   }
+
+  @Get(':username')
+  @ApiResponse({ type: UserDto, status: HttpStatus.OK })
+  async getProfileForTalent(@Param('username') username: string): Promise<UserDto> {
+
+    let user = await this.userService.findOneAsync(username);
+
+    if(user == null){
+      throw new NotFoundException('User not found');
+    }
+
+    if (user.tracks.length > 0)
+      user = await user.populate('tracks').execPopulate();
```

Comment:
We can utilize typescript's [`Pick`](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys) here to just pick non-sensitive fields.